app/models/doi.rb

```diff
@@ -878,6 +878,15 @@ def self.delete_test_dois(from_date: nil)
     end
   end
 
+  #to be used after DOIS transfer to another RA
+  def self.delete_dois_by_prefix(prefix)
+    collection.where("doi LIKE ?", prefix).find_each do |d|
```

Comment:
This is a very expensive query (using activerecord, using like, and repeating this many times). I suggest using Elasticsearch for this.